arbitrum-client/src/helpers.rs

```diff
@@ -114,3 +115,44 @@ pub fn parse_shares_from_process_match_settle(
         Err(ArbitrumClientError::BlinderNotFound)
     }
 }
+
+/// Parses wallet shares from the calldata of a `settleOnlineRelayerFee` call
+pub fn parse_shares_from_settle_online_relayer_fee(
+    calldata: &[u8],
+    public_blinder_share: Scalar,
+) -> Result<SizedWalletShare, ArbitrumClientError> {
+    let call = settleOnlineRelayerFeeCall::decode(calldata, true /* validate */)
+        .map_err(|e| ArbitrumClientError::Serde(e.to_string()))?;
+
+    let valid_relayer_fee_settlement_statement =
+        deserialize_calldata::<ContractValidRelayerFeeSettlementStatement>(
+            &call.valid_relayer_fee_settlement_statement.into(),
+        )?;
+
+    // The blinder is expected to be the last public wallet share
+    let sender_public_blinder_share =
+        valid_relayer_fee_settlement_statement.sender_updated_public_shares.last().unwrap();
+
+    // The blinder is expected to be the last public wallet share
+    let recipient_public_blinder_share =
+        valid_relayer_fee_settlement_statement.recipient_updated_public_shares.last().unwrap();
+
+    let target_share = public_blinder_share.inner();
+    if sender_public_blinder_share == &target_share {
+        let mut shares = valid_relayer_fee_settlement_statement
+            .sender_updated_public_shares
+            .into_iter()
+            .map(Scalar::new);
+
+        Ok(SizedWalletShare::from_scalars(&mut shares))
+    } else if recipient_public_blinder_share == &target_share {
+        let mut shares = valid_relayer_fee_settlement_statement
+            .recipient_updated_public_shares
+            .into_iter()
+            .map(Scalar::new);
+
+        Ok(SizedWalletShare::from_scalars(&mut shares))
+    } else {
+        Err(ArbitrumClientError::BlinderNotFound)
+    }
+}
```

Comment:
Could clean this up as:
```suggestion
    let sender_shares = &valid_relayer_fee_settlement_statement.sender_updated_public_shares;
    let sender_public_blinder_share = sender_shares.last().unwrap();

    // The blinder is expected to be the last public wallet share
    let recipient_shares = &valid_relayer_fee_settlement_statement.recipient_updated_public_shares
    let recipient_public_blinder_share = recipient_shares.last().unwrap();

    let target_share = public_blinder_share.inner();
    let selected_shares = if sender_public_blinder_share == &target_share {
        sender_shares
    } else if recipient_public_blinder_share == &target_share {
        recipient_shares
    } else {
        return Err(ArbitrumClientError::BlinderNotFound);
    }
    
    let mut share_iter = selected_shares.into_iter().map(Scalar::new);
    Ok(SizedWalletShare::from_scalars(&mut shares))
}
```